src/jdk.compiler/share/classes/com/sun/tools/javac/util/StringNameTable.java

```diff
@@ -62,7 +62,12 @@ public StringNameTable(Names names, int initialCapacity, boolean intern) {
 
     @Override
     public Name fromString(String string) {
-        return this.nameMap.computeIfAbsent(string, s -> new NameImpl(this, intern ? s.intern() : s));
+        Name name = nameMap.get(string);
+        if (name == null) {
+            name = new NameImpl(this, intern ? string.intern() : string);
+            nameMap.put(string, name);
```

Comment:
Since we are interning, can we use the possibly-interned result as map key when we put the name? This will be another bug fixed by desugaring.

Reply:
Nice catch! I don't think we use the interning today since performance was subpar, and this might be a contributing factor. Re-running the micros.

Reply:
No significant difference on the micros. Theoretically reduces the retained set during a compilation, though, so can't hurt.